joint_trajectory_controller/include/joint_trajectory_controller/tolerances.hpp

```diff
@@ -0,0 +1,320 @@
+///////////////////////////////////////////////////////////////////////////////
+// Copyright (C) 2013, PAL Robotics S.L.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//   * Redistributions of source code must retain the above copyright notice,
+//     this list of conditions and the following disclaimer.
+//   * Redistributions in binary form must reproduce the above copyright
+//     notice, this list of conditions and the following disclaimer in the
+//     documentation and/or other materials provided with the distribution.
+//   * Neither the name of PAL Robotics S.L. nor the names of its
+//     contributors may be used to endorse or promote products derived from
+//     this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//////////////////////////////////////////////////////////////////////////////
+
+/// \author Adolfo Rodriguez Tsouroukdissian
+
+#ifndef JOINT_TRAJECTORY_CONTROLLER_TOLERANCES_H
+#define JOINT_TRAJECTORY_CONTROLLER_TOLERANCES_H
+
+// C++ standard
+#include <cassert>
+#include <cmath>
+#include <string>
+#include <vector>
+
+// ROS
+#include <rclcpp/rclcpp.hpp>
+
+// ROS messages
+#include "control_msgs/action/follow_joint_trajectory.hpp"
+
+namespace joint_trajectory_controller
+{
+
+/**
+ * \brief Trajectory state tolerances for position, velocity and acceleration variables.
+ *
+ * A tolerance value of zero means that no tolerance will be applied for that variable.
+ */
+struct StateTolerances
```

Comment:
initializer lists for the win (C++14 should allow both default value initializers and the default brace-enclosed constructor to work)

```suggestion
struct StateTolerances
{
  double position = 0.0;
  double velocity = 0.0;
  double acceleration = 0.0;
};
```

Reply:
@bmagyar I don't fully understand what your suggestion here is. Isn't there already a default initialization?

Reply:
There wasn't used to be but since the change didn't happen on this line, github failed to pick it up :)